src/components/Proposal/Votes/index.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  TextCenter,
+  SidebarRow,
+  Vote,
+  AmountInput,
+  PositiveSummary,
+  NegativeSummary,
+  SummaryTotal,
+  SummaryDetails,
+  ActionButton,
+} from '../styles';
+
+import { useContext } from 'contexts';
+import { bnum, toPercentage } from 'utils';
+
+const Votes = () => {
+  const {
+    context: { daoStore, configStore, providerStore, daoService },
+  } = useContext();
+
+  //State
+  const [decision, setDecision] = useState(null);
+  const [votePercentage, setVotePercentage] = useState(0);
+
+  // We should get the ID in another way
+  const proposalId = useLocation().pathname.split('/')[3];
+  console.debug('Proposal ID in Vote: ', proposalId);
+
+  const proposal = daoStore.getProposal(proposalId);
+  const proposalEvents = daoStore.getProposalEvents(proposalId);
+  const { account } = providerStore.getActiveWeb3React();
+
+  let votedAmount = bnum(0);
+
+  proposalEvents.votes.map(vote => {
+    if (vote.voter === account) {
+      votedAmount = bnum(vote.amount);
+    }
+  });
+
+  console.log('prop events', proposalEvents);
+  let positiveVotesCount = proposalEvents.votes.filter(
+    vote => vote.vote.toString() === '1'
+  ).length;
+  let negativeVotesCount = proposalEvents.votes.filter(
+    vote => vote.vote.toString() === '2'
+  ).length;
+
+  const {
+    userRep: userRepAtProposalCreation,
+    totalSupply: totalRepAtProposalCreation,
+  } =
+    configStore.getActiveChainName().indexOf('arbitrum') > -1
+      ? daoStore.getRepAt(account, proposal.creationEvent.l2BlockNumber)
+      : daoStore.getRepAt(account, proposal.creationEvent.l1BlockNumber);
+
+  const repPercentageAtCreation = toPercentage(
+    userRepAtProposalCreation.div(totalRepAtProposalCreation)
+  ).toFixed(2);
+
+  const positiveVotes = toPercentage(
+    proposal.positiveVotes.div(totalRepAtProposalCreation)
+  ).toFixed(2);
+
+  const negativeVotes = toPercentage(
+    proposal.negativeVotes.div(totalRepAtProposalCreation)
+  ).toFixed(2);
+
+  if (Number(repPercentageAtCreation) > 0 && votePercentage === 0) {
+    setVotePercentage(Number(repPercentageAtCreation));
+  }
+
+  const { finishTime } = daoStore.getProposalStatus(proposalId);
+  const votingParameters = daoStore.getVotingParametersOfProposal(proposalId);
+  const finishTimeReached = finishTime.toNumber() < moment().unix();
+
+  // Events Handlers
+  const onVoteValueChange = event => {
+    setVotePercentage(
+      event.target.value < repPercentageAtCreation
+        ? event.target.value
+        : repPercentageAtCreation
+    );
+  };
+
+  const submitVote = function (decision) {
+    const repAmount = totalRepAtProposalCreation
+      .times(bnum(votePercentage))
+      .div('100')
+      .toFixed(0);
+    daoService.vote(decision, repAmount, proposalId);
+    setDecision(null);
+  };
+
+  return (
+    <>
+      <SidebarRow>
+        <strong>
+          Votes <Question question="4" />
+        </strong>
+      </SidebarRow>
+      <SidebarRow>
+        <PositiveSummary>
```

Comment:
Positive and negative summary components here are very similar. Create a more generic component so that it can be reused here and cuts down on LOC